jenkins/internal-ci/generic/branch/cortx-cc.groovy

```diff
@@ -0,0 +1,208 @@
+        release_dir = "/mnt/bigstorage/releases/cortx"
+        release_tag = "last_successful_prod"
+        build_upload_dir = "$release_dir/components/github/$branch/$os_version/$env/$component"
+    }
+    
+    options {
+        timeout(time: 120, unit: 'MINUTES')
+        timestamps()
+        ansiColor('xterm') 
+        disableConcurrentBuilds()  
+    }
+
+    stages {
+        stage('Checkout cortx-cc') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                dir ('cortx-cc') {
+                    checkout([$class: 'GitSCM', branches: [[name: "${branch}"]], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CloneOption', depth: 0, noTags: false, reference: '', shallow: false, timeout: 15], [$class: 'SubmoduleOption', disableSubmodules: false, parentCredentials: true, recursiveSubmodules: true, reference: '', trackingSubmodules: false, timeout: 15]], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', refspec: '+refs/heads/main:refs/remotes/origin/main', url: "https://github.com/Seagate/cortx-cc"]]])
+                }
+            }
+        }
+    
+        stage('Install Dependencies') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+
+                sh label: 'Install cortx-cc pre-requisites', script: '''
+                    yum -y install python3 python3-devel facter yum-utils
+                    yum-config-manager --add-repo https://rpm.releases.hashicorp.com/RHEL/hashicorp.repo
+                    yum -y install consul-1.9.1
+                '''
+
+                sh label: 'Install cortx-prereq', script: '''
+                    CORTX_UTILS_REPO_OWNER=$(echo $CORTX_UTILS_URL | cut -d "/" -f4)
+                    yum erase python36-PyYAML -y
+                    yum install -y gcc python3 python3-pip python3-devel python3-setuptools openssl-devel libffi-devel python3-dbus
+                    pip3 install  --no-cache-dir --trusted-host cortx-storage.colo.seagate.com -i http://cortx-storage.colo.seagate.com/releases/cortx/third-party-deps/python-deps/python-packages-$version-latest/ -r https://raw.githubusercontent.com/Seagate/cortx-utils/${branch}/py-utils/python_requirements.txt -r https://raw.githubusercontent.com/Seagate/cortx-utils/${branch}/py-utils/python_requirements.ext.txt
+                    rm -rf /etc/pip.conf
+                '''
+
+                sh label: 'Install Dependencies', script: '''
+                    set +x
+                    yum-config-manager --add-repo=http://cortx-storage.colo.seagate.com/releases/cortx/github/main/$os_version/$release_tag/cortx_iso/
+                    yum-config-manager --save --setopt=cortx-storage*.gpgcheck=1 cortx-storage* && yum-config-manager --save --setopt=cortx-storage*.gpgcheck=0 cortx-storage*
+                    yum clean all;rm -rf /var/cache/yum
+                    yum install cortx-py-utils cortx-motr{,-devel} -y --nogpgcheck
```

Comment:
Do we need to build cortx-cc for each cortx-motr change. Please get this confirmed.

Reply:
@shailesh-vaidya, cortx-cc require cortx-motr package as a pre-requisite so it make sense to trigger cortx-cc build after cortx-motr build gets complete right?

Reply:
Yes. the only thing is this will increase Motr build time. We can use last_succsful motr packages also. This need to be confirmed with the Cluster Coordinator team.

Reply:
Added last_successful repo config for motr